### DISCUSSION TITLE
Uvicorn in a kubernetes environment

### DISCUSSION THREAD

**Discussion by m-vellinga** [Kludex/uvicorn#1148] [Q&A] [answered]
Hi!

I'm trying to understand what the added benefit would be from still adding gunicorn in the deployment when your code runs on kubernetes instead of just using plain uvicorn. The [docs](https://www.uvicorn.org/deployment/#using-a-process-manager) explain the benefits/needs for a process manager. But is that still relevant when your code runs on kubernetes?

Looking forward to your train of thought.

**Comment by ediskandarov-cur** [comment]
Hi @m-vellinga 

The [documentation](https://www.uvicorn.org/deployment/#using-a-process-manager) says that uvicorn when used as a process manager *does not provide any process monitoring*.

That might be a disadvantage compared to gunicorn.

**Comment by Kludex** [comment, marked as answer]
Gunicorn will add another layer of complexity. The only goal of gunicorn is to manage the workers (uvicorn), and on K8s you have the management of pods, so you're shifting the place where you manage "things". If you wish so, you can use a single uvicorn worker, and you don't need to use Gunicorn.

The current documention needs to be updated to talk a little bit more about this subject.

**Reply by m-vellinga** [reply]
Thanks for answering my question. That confirms what I expected.

**Reply by rushilsrivastava** [reply]
For anyone stumbling on this discussion in the future, #303 is also worth checking out
